src/ape/contracts/base.py

```diff
@@ -463,6 +463,36 @@ def __len__(self):
         logs = self.provider.get_contract_logs(self.log_filter)
         return sum(1 for _ in logs)
 
+    def __call__(self, *args: Any, **kwargs: Any) -> MockContractLog:
+        # Create a dictionary from the positional arguments
+        event_args = dict(zip((input.name for input in self.abi.inputs), args))
+
+        # Check for overwrites by looking for the intersection of the two key sets
+        if overlapping_keys := set(event_args.keys()) & set(kwargs.keys()):
+            raise ValueError(f"Overlapping keys found in arguments: {overlapping_keys}")
```

Comment:
(optional) comma-join them to look nicer
```suggestion
            raise ValueError(f"Overlapping keys found in arguments: '{', '.join(overlapping_keys)}'.")
```